pkg/odo/cli/logs/logs.go

```diff
@@ -79,14 +92,42 @@ func (o *LogsOptions) Complete(cmdline cmdline.Cmdline, args []string) error {
 }
 
 func (o *LogsOptions) Validate() error {
+	if o.devMode && o.deployMode {
+		return errors.New("pass only one of --dev or --deploy flags; pass no flag to see logs for both modes")
+	}
 	return nil
 }
 
 func (o *LogsOptions) Run(ctx context.Context) error {
-	containersLogs, err := o.clientset.LogsClient.DevModeLogs(o.componentName, o.Context.GetProject())
+	var logMode logsMode
+	var err error
+	var containersLogs []map[string]io.ReadCloser
+
+	if o.devMode {
+		logMode = DevMode
+	} else if o.deployMode {
+		logMode = DeployMode
+	}
+
+	switch logMode {
+	case DevMode:
+		containersLogs, err = o.clientset.LogsClient.GetLogsForMode(odolabels.ComponentDevMode, o.componentName, o.Context.GetProject())
+	case DeployMode:
+		containersLogs, err = o.clientset.LogsClient.GetLogsForMode(odolabels.ComponentDeployMode, o.componentName, o.Context.GetProject())
+	default:
+		containersLogs, err = o.clientset.LogsClient.GetLogsForMode(odolabels.ComponentAnyMode, o.componentName, o.Context.GetProject())
```

Comment:
You could compute the mode first, then call the function only once with this computed mode